task_processing/plugins/kubernetes/kube_client.py

```diff
@@ -153,3 +153,33 @@ def create_pod(
 
         logger.info(f"Ran out of retries attempting to create {pod.metadata.name}.")
         return False
+
+    def get_pod(
+        self, namespace: str, pod_name: str, attempts: int = DEFAULT_ATTEMPTS,
+    ) -> Optional[V1Pod]:
+        while attempts:
+            try:
+                pod = self.core.read_namespaced_pod(
+                    namespace=namespace, field_selector=f'metadata.name={pod_name}'
+                ).items
+                if not pod:
+                    # XXX: We do not currently distinguish between not finding a pod and hitting api
+                    #      exception when fetching pod
```

Comment:
Yeah. I added a log line when we explicitly dont' find it so at least there's that, but otherwise downstream code won't know the difference -- is that a big issue?

Reply:
ah - it's probably worth re-raising the ApiException instead of always returning None

Reply:
Alright I've gone this route though I wasn't clear what we wanted to do on the reconcile() side -- should we try to just remove the pod from task_metadata or leave it be at that point in `TASK_UNKNOWN` forever? 

Will we be trying to reconcile all known running pods on some interval?